### PR TITLE
Add inventory check function

### PR DESCRIPTION
**Pull request by liubo0127**
@sykp241095 PTAL

**For example:**
image

### CODE REVIEW

File: scripts/inventory_check.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                  .format(','.join(tidb_conf_conflict)))
+    else:
+        if tikv_conf_conflict:
+            print('''
+    TiKV port or deployment directory conflicts on {} machine.
+    Please recheck the port, status_port, deploy_dir or other configuration in inventory.ini.'''
+                  .format(','.join(tikv_conf_conflict)))
+        else:
+            print('Check ok.')
+
```
Comment (sykp241095) [inline scripts/inventory_check.py]:
```python
if tidb_conflictions:
    print 'TiDB port or deployment directory conflicts on {} machine.'
if tikv_conflictions:
    print 'TiKV port or deployment directory conflicts on {} machine.'

if tidb_conflictions or tikv_conflictions:
    print 'Please recheck the port, status_port, deploy_dir or other configuration in inventory.ini.'

File: scripts/inventory_check.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        if ip in tidb_servers:
+            tidb_servers[ip].append([tidb_port, tidb_status_port, deploy_dir])
+        else:
+            tidb_servers[ip] = [[tidb_port, tidb_status_port, deploy_dir]]
+
+    for tikv in tikv_nodes:
+        var = vars.get_vars(host=inv.get_host(hostname=str(tikv)))
+        ip = var['ansible_host'] if 'ansible_host' in var else var['inventory_hostname']
+        tikv_port = var['tikv_port'] \
+            if 'tikv_port' in var else 4000
```
Comment (qrr1995) [inline scripts/inventory_check.py]: `tikv_port = var.get('tikv_port', 4000)` is more simple and readable,the same below.

File: scripts/inventory_check.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    tidb_conf_conflict = check_conflict(tidb_servers)
+    tikv_conf_conflict = check_conflict(tikv_servers)
+    if tidb_conf_conflict:
+        print('\n    TiDB port or deployment directory conflicts on {} machine.'
+              .format(','.join(tidb_conf_conflict)))
+
+    if tikv_conf_conflict and not tidb_conf_conflict:
+        print('\n    TiKV port or deployment directory conflicts on {} machine.'
+              .format(','.join(tikv_conf_conflict)))
+    elif tikv_conf_conflict and tidb_conf_conflict:
```
Comment (qrr1995) [inline scripts/inventory_check.py]: These `if-elif` conditions seems meaningless?Just same as:
```
if tikv_conf_conflict:
    print "something"
```

File: scripts/inventory_check.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    return [tidb_servers, tikv_servers]
+
+
+def check_conflict(server_list):
+    conflict_ip = []
+    isContinue = False
+    for ip, nodes in server_list.iteritems():
+        length = len(nodes)
+        if length > 1:
```
Comment (qrr1995) [inline scripts/inventory_check.py]: Meanningless condition, `for index_i in range(0)` is OK which did nothing.

File: scripts/inventory_check.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+            for index_i in range(length - 1):
+                for index_j in range(index_i + 1, length):
```
Comment (qrr1995) [inline scripts/inventory_check.py]: Too nested, it's neater to implement a flat structure:
```
port_list = [var_triple[0] for var_triple in nodes] + [var_triple[1] for var_triple in nodes] 
dir_list = [var_triple[3] for var_triple in nodes]

# duplicate port in a node means conflict.
if len(set(port_list)) < len(port_list):
    conflict_ip.append(ip)
elif len(set(dir_list)) < len(dir_list):
    conflict_ip.append(ip)
else:
    print 'This IP is ok!'
```

**Pull request merged**